### PR TITLE
Add der_spki() methods for all supported key types

### PR DESCRIPTION
**Pull request by jschanck**
The `getPublicKey()` method in WebAuthn Level 3 outputs a DER encoded SubjectPublicKeyInfo, and it has [the following requirement](https://w3c.github.io/webauthn/#sctn-public-key-easy):
> User agents MUST be able to return a non-null value for [getPublicKey()](https://w3c.github.io/webauthn/#dom-authenticatorattestationresponse-getpublickey) when the [credential public key](https://w3c.github.io/webauthn/#credential-public-key) has a [COSEAlgorithmIdentifier](https://w3c.github.io/webauthn/#typedefdef-cosealgorithmidentifier) value of:
>
>    -7 (ES256), where [kty](https://tools.ietf.org/html/rfc9052#name-cose-key-common-parameters) is 2 (with uncompressed points) and [crv](https://tools.ietf.org/html/rfc9053#name-double-coordinate-curves) is 1 (P-256).
    -257 (RS256).
    -8 (EdDSA), where [crv](https://tools.ietf.org/html/rfc9053#name-double-coordinate-curves) is 6 (Ed25519).

This PR adds der_spki() methods for RS256 and Ed25519 and improves the serialization tests for all key types.

It turns out that #292 did not resolve all of the issues with the CBOR encoding of OKP keys (see also: Firefox Bug [1852812](https://bugzilla.mozilla.org/show_bug.cgi?id=1852812) and references therein), so I've included a patch for that as well.

### CODE REVIEW

File: src/crypto/der.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+pub fn integer(val: &[u8]) -> Result<Vec<u8>> {
+    // trim leading zeros, leaving a single zero if the input is the zero vector.
+    let mut val = val;
+    while val.len() > 1 && val[0] == 0 {
+        val = &val[1..];
+    }
+
+    let mut out = Vec::with_capacity(MAX_TAG_AND_LENGTH_BYTES + 1 + val.len());
+    if !val.is_empty() && val[0] & 0x80 != 0 {
```
Comment (mozkeeler) [inline src/crypto/der.rs]: Should empty values even be allowed in this function?

File: src/crypto/mod.rs [hunk 2]
```diff
@@ -726,37 +716,25 @@ impl COSEEC2Key {
...
-        spki.push(0x0); // no trailing zeros
-        spki.push(0x04); // SEC 1 encoded uncompressed point
-        spki.extend_from_slice(&self.x);
-        spki.extend_from_slice(&self.y);
-
-        Ok(spki)
+        der::sequence(&[
+            // algorithm: AlgorithmIdentifier
+            &der::sequence(&[
+                /* algorithm */
```
Comment (mozkeeler) [inline src/crypto/mod.rs]: nit: let's not mix comment styles if we don't have to

**Review by mozkeeler — APPROVED** [review]
LGTM!

File: src/crypto/der.rs (hunk 1 above)
Comment (jschanck) [reply src/crypto/der.rs]: Good catch. X.690 says "contents octets shall consist of one or more octets".

**Pull request merged**